This pull request is to merge rspec_tic_tac_toe branch to the Development branch

henatan99 opened the pull request:
### **Testing Tic-tac-toe project using rspec.** 

- the first step of this project is installing rspec
- After installation rspec initialized in the project root directory 
- Test files are created for each class files: player_spec.rb, board_spec.rb and tictactoe_spec.rb, in the spec folder
- Positive and negative scenario examples are used for all the methods in the classes

henatan99 committed:
Initialize init in root directory
- Creates tictactoe_spec.rb file in spec folder

henatan99 committed:
Adds test exampes for all methods in all classes

henatan99 committed:
Fix rubocop offenses
- Avoids unecessary assignment operators
- Fix trailing space offenses
- Fix too long line offenses

peterrobert commented on spec/board_spec.rb:
```diff
@@ -0,0 +1,30 @@
+    # it " returns arg[1]" do
+    #     board1 = Board.new
+    #     expect(board1.update_board(2)).to raise_error(ArgumentError)
+    # end
```
Commented code should be deleted. This will make your work clean.
<img width="792" alt="messy" src="https://user-images.githubusercontent.com/32361963/101313800-1f6e2b00-3868-11eb-8daf-eeaf5de2b997.png">

peterrobert commented on spec/player_spec.rb:
```diff
@@ -0,0 +1,30 @@
+    # end
+    # it "returns Argument error when object is attempted to be created with no argument" do
+    #     expect(player4).to raise_error(ArgumentError)
+    # end
```
Kindly remove all the commented code in your project

peterrobert commented on spec/player_spec.rb:
```diff
@@ -0,0 +1,30 @@
+  let(:player2) { Player.new('Nikola') }
+  let(:player3) { Player.new('X') }
+  let(:player4) { Player.new }
+  describe 'initialize' do
+    it 'returns player object pointer when a player object is created by name and symbol' do
+      expect(player1).to be_kind_of(Player)
+    end
```
Kindly you should test for both positive and negative scenarios for instance:
`returns player object pointer when a player object is created by name and symbol` :+1:  But what if the player object doesn't have a name or a symbol. You should test for that failure too.

**Nb:** Kindly do so for all your tests. Make sure you check for both positive and negative scenarios.

![commonsense](https://user-images.githubusercontent.com/32361963/101314249-29dcf480-3869-11eb-8103-5e889014fd28.png)

peterrobert commented on spec/tictactoe_spec.rb:
```diff
@@ -0,0 +1,232 @@
+    end
+    # it "returns Argument error when wrong number of argumets is given" do
+    #     expect(game.create_players('Henok', 'Nikola', 'X')).to raise_error(ArgumentError)
+    # end
```
Kindly delete commented code

peterrobert reviewed (changes requested):
### CHANGES REQUIRED :recycle: :stop_sign: 
HELLO, @henatan99  AND @zdrale VERY NICE PROGRESS. I LOVE THAT YOU HAVE CREATED TESTS FOR ALL OF YOUR METHODS. JUST NEED SOME FEW CHANGES AND WE SHALL BE GOOD TO GO. KINDLY CHECK BELOW :arrow_down: 

- [ ] There are some rubocop errors in your project. Kindly fix all the errors. :arrow_down: 

![rubocop](https://user-images.githubusercontent.com/32361963/101314684-0f574b00-386a-11eb-9853-7ca9b834f65a.png)

**summary**

- [ ] Kindly test for both the positive and negative scenarios. Just like I have described it below.

- [ ] Kindly remove all the commented code in your project.

- [ ] Your pull request message is descriptive. However, you should add a title and small description points telling us what you have worked on in this project

### [OPTIONAL]

- [ ] Using `require_relative` is usually in preference compared to just using `require` to link files in your project. Kindly try and use require_relative/


### if you have any question kindly reach me on slack @peterrobert


### HAPPY CODING :computer:

PhillipUg reviewed (approved):
# Status: Project Approved! 🥇 
Congratulations on implementing all project requirements and getting this project approved. :+1:

You can merge this branch into the development branch when you're ready and go on to your next project.
:bulb:  Do not forget to submit your project completion form after merging this branch.

All the best team,

For any assistance, you can contact me on slack as **Phillip Musiime**

henatan99 committed:
Fix TSE's comments
- Adds negative scenarios
- Adds more description on PR
- Removes inline comments
- Fix rubocop offesnses offenses
